index/index.go

```diff
@@ -0,0 +1,137 @@
+
+// Init will initialize the IndexResource structure and will return it.
+// It will take indexString as a paramenter.
+func Init(indexString string, directory *paths.Path) *Resource {
+	if directory == nil {
+		log.Fatalf("configuration directory not provided")
+	}
+	if !directory.Exist() {
+		err := directory.Mkdir()
+		if err != nil {
+			log.Fatalf("cannot create config dir: %s", err)
+		}
+		err = hideFile(directory.String())
+		if err != nil {
+			log.Printf("cannot make config dir hidden: %s", err)
+		}
+	}
+	indexParsed, err := url.Parse(indexString)
+	if err != nil {
+		log.Fatalf("cannot parse provided index: %s", indexString)
+	}
+
+	indexFile := path.Base(indexParsed.Path) // == package_index.json
+	signatureFile := indexFile + ".sig"
+
+	var ir = Resource{
+		IndexURL:       *indexParsed,
+		IndexFile:      *directory.Join(indexFile),
+		IndexSignature: *directory.Join(signatureFile),
+	}
+
+	if !ir.IndexFile.Exist() || time.Since(ir.LastRefresh) > 1*time.Hour {
+		// Download the file again and save it
+		if err := ir.DownloadAndVerify(); err != nil {
+			log.Fatalf("cannot download index: %s", err)
+		}
+	}
```

Comment:
```suggestion
	if err := ir.DownloadAndVerify(); err != nil {
		log.Fatalf("cannot download index: %s", err)
	}
```

The `Init` function is called only once during the startup of the application. I'd simplify this by always downloading the index the first time.